build/separate.md

```diff
@@ -81,3 +81,11 @@ install them into `$HOME/install`, then a configure command line for this
 looks something like this:
 
     LDFLAGS="-Wl,-rpath,$HOME/install/lib" ./configure ...
+
+If your system supports the runpath form of rpath it is often better to use
+that instead because it can be overridden by the `LD_LIBRARY_PATH` environment
+variable. It may also prevent libtool bugs when testing in-tree builds of curl,
+since then libtool can use `LD_LIBRARY_PATH`. To use an rpath as a runpath add
+the linker flag `-Wl,--enable-new-dtags` like this:
+
+    LDFLAGS="-Wl,-rpath,$HOME/install/lib -Wl,--enable-new-dtags" ./configure ...
```

Comment:
Jeffrey Walton pointed out this interesting link on the mailing list:

https://stackoverflow.com/a/55283740

It highlights the fact that gcc's `-Wl,-rpath=` option already in fact set the `runpath` property if you use a new gcc. i verified this with gcc 13.2.0 on Debian.

Somewhat more explained here: https://rohgarg.github.io/notes/2020/02/20/library-search-order.html.

Thus, this section should probably emphasize that this is only valid for "old" linkers. I have not managed to figure out exactly when this change was done.

Reply:
I don't think there's enough info to say that for certain that only older linkers set RPATH by default. I pushed a change that says newer linkers may use it:

---
Newer linkers may use the runpath form of rpath by default when rpath is specified but others need an additional linker flag `-Wl,--enable-new-dtags` like this:

    LDFLAGS="-Wl,-rpath,$HOME/install/lib -Wl,--enable-new-dtags" ./configure ...

---

Reply:
The way you phrase this makes it sound as if this is the preferred way to do it with current toolchains, while in reality is a way to do it only with **old** compilers/linkers. I think this should rather stress that part. The already document way sets runpath with current toolsets.